plugin/src/main/java/com/exadel/aem/toolkit/plugin/handlers/common/AllowedChildrenHandler.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exadel.aem.toolkit.plugin.handlers.common;
+
+import com.exadel.aem.toolkit.api.annotations.main.ac.AllowedChildren;
+import com.exadel.aem.toolkit.api.annotations.meta.Scopes;
+import com.exadel.aem.toolkit.api.handlers.Handler;
+import com.exadel.aem.toolkit.api.handlers.Source;
+import com.exadel.aem.toolkit.api.handlers.Target;
+import com.exadel.aem.toolkit.plugin.utils.DialogConstants;
+import com.google.gson.Gson;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class AllowedChildrenHandler implements Handler {
+
+    private static final Gson GSON = new Gson();
+
+    @Override
+    public void accept(Source source, Target target) {
+        source.tryAdaptTo(AllowedChildren[].class).ifPresent(adaptation -> populatePolicies(adaptation, target));
+    }
+
+    private void populatePolicies(AllowedChildren[] rules, Target target) {
```

Comment:
Please consider declaring methods of this class that do not consume nor mutate the fields of this class as `static`